pkg/imgpkg/cmd/copy_repo_src_test.go

```diff
@@ -273,10 +276,81 @@ func TestToTarImage(t *testing.T) {
 		imageTarPath := filepath.Join(os.TempDir(), "bundle.tar")
 		defer os.Remove(imageTarPath)
 
-		err := subject.CopyToTar(imageTarPath)
+		err := subject.CopyToTar(imageTarPath, false)
 		require.NoError(t, err)
 
-		assert.Contains(t, stdOut.String(), "Warning: '--include-non-distributable-layers' flag provided, but no images contained a non-distributable layer.\n")
+		require.Contains(t, stdOut.String(), "Warning: '--include-non-distributable-layers' flag provided, but no images contained a non-distributable layer.\n")
+	})
+
+	t.Run("When copy to tar fails the first time but second call with resume completes successfully", func(t *testing.T) {
+		numberOfRequests := 0
+		var failedDigest regv1.Hash
+		var layersInTar []regv1.Layer
+		fakeRegistry.WithCustomHandler(func(writer http.ResponseWriter, request *http.Request) bool {
+			matched, err := regexp.MatchString("/v2/.+/blobs", request.URL.Path)
+			require.NoError(t, err)
+
+			if matched && request.Method == "GET" {
+				parts := strings.Split(request.URL.Path, "/")
+				sha := parts[len(parts)-1]
+				hash, err := regv1.NewHash(sha)
+				// This loop ensures that if a layer is in the tar already we can return gibberish\
+				// because imgpkg is not going to use this information
+				for _, layer := range layersInTar {
+					digest, err := layer.Digest()
+					require.NoError(t, err)
+					if hash.String() == digest.String() {
+						size, err := layer.Size()
+						require.NoError(t, err)
+						bs := make([]byte, size)
+						writer.Write(bs)
+						return true
+					}
+				}
+
+				numberOfRequests++
+				// This if statement makes sure that in the 5th layer that imgpkg tries to retrieve some gibberish
+				// is returned to simulate a failure in communication
+				if numberOfRequests == 5 {
+					require.NoError(t, err)
+					if !fakeRegistry.IsConfigBlobLayer(hash) {
+						layer, err := fakeRegistry.Layer(hash)
+						require.NoError(t, err)
+						size, err := layer.Size()
+						require.NoError(t, err)
+						writer.WriteHeader(http.StatusOK)
+						bs := make([]byte, size)
+						writer.Write(bs)
+						failedDigest = hash
+						return true
+					}
+
+					// Found configuration layer, it doesn't count as content blob
+					numberOfRequests--
+				}
+			}
+
+			return false
+		})
+
+		imageTarPath := filepath.Join(os.TempDir(), "img.tar")
+		defer os.Remove(imageTarPath)
```

Comment:
if we are hard coding temp location, then we should remove it before it use. additionally lets namespace the name e.g. imgpkg-test-img.tar